src/lib/json-rpc-router.ts

```diff
@@ -0,0 +1,37 @@
+import type { Request } from 'express';
+import type { Readable } from 'readable-stream';
+import type { JsonRpcRequest, JsonRpcResponse } from './json-rpc.js';
+
+type HttpRequest = {
+  contentType: 'application/octet-stream' | 'multipart/form-data';
+  request: Request;
+};
+
+export type RequestContext = {
+  transport?: 'http' | 'ws';
+} & (HttpRequest | { [K in keyof HttpRequest]?: never; });
```

Comment:
does this mean "a `HttpRequest` or an object that contains anything except for the keys in `HttpRequest`"?  if so, what are we really gaining here by doing this?

Reply:
https://github.com/TBD54566975/dwn-server/pull/13#discussion_r1177312359